src/monmove.cpp

```diff
@@ -99,10 +99,27 @@ bool monster::is_immune_field( const field_type_id fid ) const
     return Creature::is_immune_field( fid );
 }
 
+static bool z_is_valid( int z )
+{
+    return z >= -OVERMAP_DEPTH && z <= OVERMAP_HEIGHT;
+}
+
 bool monster::can_move_to( const tripoint &p ) const
 {
     const bool can_climb = has_flag( MF_CLIMBS ) || has_flag( MF_FLIES );
 
+    if( p.z > pos().z && z_is_valid( pos().z ) ) {
+        if( !g->m.has_flag( TFLAG_GOES_UP, pos() ) && g->m.ter( p ) != t_hole ) {
```

Comment:
Maybe introduce a new flag instead of hardcoding to `t_hole`? To allow other terrain/furniture types in the future, like a hatch or a trapdoor.

Reply:
that's definitely reasonable. i think maybe i'll call it something like EMPTY_Z or some such, to denote that it's "passable" in the z direction

Reply:
update: found a flag called NO_FLOOR, but i need to do some digging to figure out the ramifications of this flag, as it wasn't readily apparent what it does.